client_with_ssl.cpp

```diff
@@ -1,17 +1,21 @@
-#include <hazelcast/client/hazelcast_client.h>
 #include <string>
+#include "model/City.h"
+#include "model/Country.h"
 
+// This is boilerplate application that configures client to connect Hazelcast
+// Cloud cluster.
+// See: https://docs.hazelcast.com/cloud/cpp-client
+
+void mapExample(hazelcast::client::hazelcast_client client);
+void nonStopMapExample(hazelcast::client::hazelcast_client client);
 int main(int argc, char **argv) {
     hazelcast::client::client_config config;
-    
     config.set_cluster_name("YOUR_CLUSTER_NAME");
     config.set_property(hazelcast::client::client_properties::CLOUD_URL_BASE, "YOUR_DISCOVERY_URL");
     config.set_property("hazelcast.client.statistics.enabled", "true");
-
     auto &cloud_configuration = config.get_network_config().get_cloud_config();
     cloud_configuration.enabled = true;
     cloud_configuration.discovery_token = "YOUR_CLUSTER_DISCOVERY_TOKEN";
-
     boost::asio::ssl::context ctx(boost::asio::ssl::context::tlsv12);
     ctx.set_verify_mode(boost::asio::ssl::verify_peer);
```

Comment:
do we need this? I could not see corresponding code at Java client. is java mutual authentication enabled by default, can you double check with @mdumandag?

Reply:
In example code mutual_authentication.cpp it is enabled explicitly like here which made me believe it is(or was) disabled by default.

Reply:
The mutual authentication is probably enabled on the member-side, see https://github.com/hazelcast/hazelcast-cloud/blob/23345fefab913d813a7ec22af4ec8e7dd81a1659/hazelcast-cloud-orchestrator-service/src/main/resources/templates/hazelcast-hz5_1.ftl#L126

Of course, it would be good to have a confirmation from the cloud team as well.

BTW @ihsandemir, you enabled the mutual authentication on the member side, to verify the clients. That's why there is no setting for it in the client side. See https://docs.hazelcast.com/hazelcast/5.1/security/tls-ssl#mutual-authentication

Reply:
it works without that line so line is omitted.